rosbag2_transport/test/rosbag2_transport/test_play_publish_clock.cpp

```diff
@@ -64,40 +59,42 @@ class ClockPublishFixture : public RosBag2PlayTestFixture
     prepared_mock_reader->prepare(messages, topic_types);
     auto reader = std::make_unique<rosbag2_cpp::Reader>(std::move(prepared_mock_reader));
 
+    auto player = std::make_shared<MockPlayer>(std::move(reader), storage_options_, play_options_);
+
+    sub_->add_subscription<test_msgs::msg::BasicTypes>("/topic1", messages.size());
```

Comment:
It turns out that for tests with timer we still need to spin player node
```suggestion
    auto player = std::make_shared<MockPlayer>(std::move(reader), storage_options_, play_options_);

    rclcpp::executors::SingleThreadedExecutor exec;
    exec.add_node(player);
    auto spin_thread = std::thread(
      [&exec]() {
        exec.spin();
      });

    sub_->add_subscription<test_msgs::msg::BasicTypes>("/topic1", messages.size());
```